app/controllers/int_api/marketplaces_controller.rb

```diff
@@ -47,7 +47,14 @@ def create
         marketplace[:id]).data
 
     # make the marketplace creator land to the admin panel
-    url = "#{marketplace[:url]}/admin"
+    url_params = {host: marketplace[:url]}
+
+    #include port separately to url_params if it was present
+    if marketplace[:url] && marketplace[:url].split(':')[2]
+      url_params.merge!(port: marketplace[:url].split(':')[2])
+    end
+
+    url = admin_getting_started_guide_url(url_params)
```

Comment:
This is better now with no string catenating and using path helper. Two small suggestions: no need to use mutable url_params here, the code feels overly complex with conditional mutation and Ruby's URI class is better first choice than string splitting whenever thinking about URI/URL parsing:

```ruby
base_url = URI.new(marketplace[:url])
url = admin_getting_started_guide_url(host: base_url.host, port: base_url.port)
```